schools_change_detection/schools_change_detection.py

```diff
@@ -0,0 +1,862 @@
+        "password": {"type": "string"},
+        "schema": {"type": "string"},
+        "table": {"type": "string"},
+    },
+    "additionalProperties": False,
+}
+FILESCHEMA = None
+OUTPUT_SCHEMAS: Dict[str, Dict[str, Union[str, Dict[str, str]]]] = {
+    "nz_facilities": {
+        "geometry": "MultiPolygon",
+        "properties": {
+            "facility_id": "int",
+            "source_facility_id": "str",
+            "name": "str",
+            "source_name": "str",
+            "use": "str",
+            "use_type": "str",
+            "use_subtype": "str",
+            "estimated_occupancy": "int",
+            "last_modified": "date",
+            "change_action": "str",
+            "change_description": "str",
+            "comments": "str",
+            "sql": "str",
+        },
+    },
+    "moe_schools": {
+        "geometry": "Point",
+        "properties": {
+            "School_Id": "int",
+            "Org_Name": "str",
+            "Add1_Line1": "str",
+            "Add1_Suburb": "str",
+            "Add1_City": "str",
+            "Org_Type": "str",
+            "Roll_Date": "date",
+            "Total": "int",
+            "Latitude": "float",
+            "Longitude": "float",
+            "change_action": "str",
+            "change_description": "str",
+        },
+    },
+}
+
+MOE_ENDPOINT = "https://catalogue.data.govt.nz/api/3/action/datastore_search_sql"
+
+MOE_SQL = 'SELECT "School_Id", "Org_Name", "Add1_Line1", "Add1_Suburb", "Add1_City", "Org_Type", \
+         "Latitude", "Longitude", "Roll_Date", "Total" FROM "20b7c271-fd5a-4c9e-869b-481a0e2453cd" \
+             ORDER BY "School_Id"'
+
+FACILITIES_FIELDS = [
+    "facility_id",
+    "source_facility_id",
+    "name",
+    "source_name",
+    "use",
+    "use_type",
+    "use_subtype",
+    "estimated_occupancy",
+    "last_modified",
+    "ST_AsGeoJSON(shape) as shape",
+]
+FACILITIES_FILTER = "WHERE use = 'School'"
+
+EPSG_2193 = pyproj.CRS("EPSG:2193")
+EPSG_4326 = pyproj.CRS("EPSG:4326")
+
+DISTANCE_FROM_FACLITY = 30
+TEEN_UNIT_DISTANCE = 100
+
+############################################
+# Exceptions, Customised classes and alias #
+############################################
+
+
+class FatalError(Exception):
+    pass
+
+
+GeoRecord = Dict[str, Dict[str, Any]]
+
+
+@dataclass
+class Source:
+    """
+    Base class for facilities and moe schools.
+
+    """
+
+    source_id: int
+    source_name: str
+    source_type: str  # good use for an enum once we work out what they should be
+    geom: str
+    occupancy: Optional[int] = None
+    change_action: Optional[str] = None
+    change_description: Optional[str] = None
+    comments: Optional[str] = None
+
+    def __eq__(self, other: "Source") -> bool:
+        results = {
+            "source_id": self.source_id == other.source_id,
+            "source_name": self.source_name == other.source_name,
+            "source_type": self.source_type == other.source_type,
+        }
```

Comment:
The "occupancy" column is not evaluated here because every school's occupancy has changed.  However, we need the option to include "occupancy" because we will want to update these values at a later stage.  It might be best for the user to decide on which columns to evaluate, including an "all" option.